File: LAST/第七題-Python 繼承/Sphere.py
from Shape import Shape

class Sphere(Shape):
    def __init__(self, radius, color):
        self.__radius = radius
        super().setColor(color)

    def setRadius(self, radius):
         self.__radius = radius
    def getRadius(self):
        return  self.__radius

    def __str__(self):
        return "Sphere"

    def getVolume(self):
        return 4 * 3.14159 * self.__radius ** 3 / 3

File: LAST/第四題-C++多型/Sphere.cpp
#include"Sphere.h"

Sphere::Sphere(double radius, string color) {
	this->radius = radius;
	setColor(color);
}

void Sphere::toString() {
	cout << "Sphere:(" << radius << "),";
}
double Sphere::getVolume() {
	return 4 * 3.14159 * pow(radius, 3) / 3;
}